selfdrive/common/util.cc

```diff
@@ -97,5 +102,85 @@ int write_file(const char* path, const void* data, size_t size, int flags, mode_
   return (n >= 0 && (size_t)n == size) ? 0 : -1;
 }
 
+std::string readlink(const std::string &path) {
+  char buff[4096];
+  ssize_t len = ::readlink(path.c_str(), buff, sizeof(buff)-1);
+  if (len != -1) {
+    buff[len] = '\0';
+    return std::string(buff);
+  }
+  return "";
+}
+
+bool file_exists(const std::string& fn) {
+  std::ifstream f(fn);
+  return f.good();
+}
+
+std::string getenv_default(const char* env_var, const char * suffix, const char* default_val) {
+  const char* env_val = getenv(env_var);
+  if (env_val != NULL){
+    return std::string(env_val) + std::string(suffix);
+  } else {
+    return std::string(default_val);
+  }
+}
+
+std::string tohex(const uint8_t *buf, size_t buf_size) {
+  std::unique_ptr<char[]> hexbuf(new char[buf_size * 2 + 1]);
+  for (size_t i = 0; i < buf_size; i++) {
+    sprintf(&hexbuf[i * 2], "%02x", buf[i]);
+  }
+  hexbuf[buf_size * 2] = 0;
+  return std::string(hexbuf.get(), hexbuf.get() + buf_size * 2);
+}
+
+std::string base_name(std::string const &path) {
+  size_t pos = path.find_last_of("/");
+  if (pos == std::string::npos) return path;
+  return path.substr(pos + 1);
+}
+
+std::string dir_name(std::string const &path) {
+  size_t pos = path.find_last_of("/");
+  if (pos == std::string::npos) return "";
+  return path.substr(0, pos);
+}
 
 }  // namespace util
+
+
+// class ExitHandler
+
+class ExitHandlerHelper {
```

Comment:
why not just put this in the ExitHandler class?

Reply:
Putting it in the header file will conflict with android_system_core/include/cutils/atomic.h

Reply:
hmm, can we just wrap the include in a namespace?

Reply:
It may not be simple

Reply:
Do we really want to go through all this effort to "hide" the std::atomic<bool>? I'd rather force the include order, since the problem will just come back if we just std::atomic<bool> somewhere else.

Reply:
yes that would be better